sample/kotlin/src/main/kotlin/com/datadog/android/sample/compose/NavigationSample.kt

```diff
@@ -129,3 +149,15 @@ internal sealed class Screen(
 internal fun <T> oneOf(vararg items: T): T {
     return items[Random.nextInt(items.size)]
 }
+
+private class DatadogLifecycleObserver : LifecycleEventObserver {
+    private var viewIsLoaded = false
+
+    @OptIn(ExperimentalRumApi::class)
+    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+        if (event == Lifecycle.Event.ON_RESUME && !viewIsLoaded) {
```

Comment:
> [!NOTE]
> Is this sample the kind of use case we want to showcase ? Tracking the `onResume` time is not relevant most of the time. We should probably instead only do what you added in the `DataListViewModel` where we actually load something from the network, which would be a better example of how to use this new API.

Reply:
Regarding your last comment, that is a nice to have in the Sample app, my intention was to add the viewLoadingTime for most of the views there. In that case I will remove the call for this view and maybe add a task to enrich the sample app. This is not the scope for this PR. @0xnm what's your opinion on that ?

Reply:
It depends on the nature of the screen, some screens may not load anything from the network/disk. But maybe to highlight that this method shouldn't be blindly used in the `onResume` we may want indeed to have it called only in the data stream.

Reply:
On the same time we could agree that there will be use cases on the customer's end where simple static views like these ones would have to be measured. In that case we will remove these use - cases from the sample app and only maintain the ones that imply an async data ?